### PR TITLE
Text transform options

### PR DESCRIPTION
**Pull request by karimnaaji**
Add `capitalize`, `lowercase` text-transform options.

### CODE REVIEW

**Comment by karimnaaji** [comment]
Right, much simpler and no conflicts

File: core/src/text/textBuffer.cpp [hunk 1]
```diff
@@ -33,18 +33,45 @@ bool TextBuffer::addLabel(const std::string& _text, Label::Transform _transform,
...
+    if (_params.transform != TextTransform::none) {
+        transformedText = _text;
         std::locale loc;
-        for (std::string::size_type i = 0; i < _text.length(); ++i) {
-            text[i] = std::toupper(_text[i], loc);
+
+        // perfom text transforms
+        switch (_params.transform) {
+            case TextTransform::capitalize:
+                transformedText[0] = toupper(_text[0], loc);
```
Comment (tallytalwar) [inline core/src/text/textBuffer.cpp]: what if a text starts with a whitespace ... possible with osm data?

Comment (karimnaaji) [inline core/src/text/textBuffer.cpp]: It would toupper on a whitespace, which I guess does nothing. Noticed it's missing `std::`.

Comment (tallytalwar) [inline core/src/text/textBuffer.cpp]: :+1:

**Comment by tallytalwar** [comment]
looks good.

**Comment by tallytalwar** [comment]
Looks good, once rebased can be merged.

**Pull request merged**